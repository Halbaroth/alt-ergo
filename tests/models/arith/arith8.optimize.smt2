(set-logic ALL)
(set-option :produce-models true)
(declare-const x Int)
(declare-const y Real)
(declare-const z Int)
(push 1)
(assert (<= x 1))
; This objective cannot be fulfilled and the optimization should stop here.
(assert (< y 0.5))
(assert (< (- 0 5) z))
(maximize x)
(maximize y)
(minimize z)
(check-sat)
(get-model)
(pop 1)
(push 1)
(assert (<= x 1))
(assert (<= y 0.5))
(assert (< (- 0 5) z))
(maximize x)
(maximize y)
(minimize z)
(check-sat)
(get-model)
(pop 1)
