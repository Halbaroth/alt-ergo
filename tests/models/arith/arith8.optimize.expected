
unknown
(
  (define-fun x () Int 1)
  (define-fun z () Int 0)
  (define-fun y () Real 0)
)

unknown
(
  (define-fun x () Int 1)
  (define-fun z () Int (- 4))
  (define-fun y () Real (/ 1 2))
)
